### PR TITLE
x-pack batch 2 instanceof pattern matching replacement

### PR DESCRIPTION
**Pull request by astefan**
Switch to Java 17 `instanceof` pattern matching for x-pack folders `x-pack/plugin/eql` through `x-pack/plugin/searchable-snapshots`.

### CODE REVIEW

**Comment by astefan** [comment]
@elasticsearchmachine run elasticsearch-ci/bwc

**Comment by astefan** [comment]
@elasticsearchmachine update branch

**Comment by astefan** [comment]
@elasticmachine update branch

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/AnalysisUtils.java [hunk 1]
```diff
@@ -76,10 +76,8 @@ private static Attribute handleSpecialFields(UnresolvedAttribute u, Attribute na
         if (named instanceof FieldAttribute fa) {
 
             // incompatible mappings
-            if (fa.field() instanceof InvalidMappedField) {
-                named = u.withUnresolvedMessage(
-                    "Cannot use field [" + fa.name() + "] due to ambiguities being " + ((InvalidMappedField) fa.field()).errorMessage()
-                );
+            if (fa.field()instanceof InvalidMappedField field) {
```
Comment (dakrone) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/AnalysisUtils.java:79]: Missing an extra space here
```suggestion
            if (fa.field() instanceof InvalidMappedField field) {
```

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManagerTests.java [hunk 2]
```diff
@@ -161,8 +161,7 @@ public void setup() throws Exception {
         when(threadPool.executor(anyString())).thenReturn(EsExecutors.DIRECT_EXECUTOR_SERVICE);
         when(client.threadPool()).thenReturn(threadPool);
         doAnswer(invocationOnMock -> {
-            if (invocationOnMock.getArguments()[0] instanceof ActionType<?>) {
-                ActionType<?> v = (ActionType<?>) invocationOnMock.getArguments()[0];
+            if (invocationOnMock.getArguments()[0]instanceof ActionType<?> v) {
```
Comment (dakrone) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManagerTests.java:164]: Another typo here
```suggestion
            if (invocationOnMock.getArguments()[0] instanceof ActionType<?> v) {
```

**Review by dakrone — APPROVED** [review]
LGTM, I left two comments about typos

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/AnalysisUtils.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/AnalysisUtils.java:79]: @dakrone thank you for reviewing and catching the typos. It's a known issue in Spotless and, for the moment, the options are:
1. Accept it
2. Don't use that syntax
3. Use `// tag:noformat` and `// end::noformat`

In this case I chose to leave it as is and, when the bug is fixed, hopefully `spotlessJavaCheck` will catch all these typos and fix them in bulk.

**Pull request merged**